app/models/badge_claim.rb

```diff
@@ -0,0 +1,15 @@
+class BadgeClaim < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :badge
+
+  validates :badge, presence: true
+
+  def redeem(user)
+    cmd = Actions::RedeemBadgeClaim.new(self, user)
+    cmd.perform
+  end
+
+  def self.create_many(badge, count)
+    BadgeClaim.create(Array.new(count) {{badge: badge}})
```

Comment:
Thanks! There are advantages to stumbling around Rails without knowing much about common practice.